featuretools/entityset/serialize.py

```diff
@@ -97,23 +104,65 @@ def write_entity_data(entity, path, format='csv', **kwargs):
-        kwargs (keywords) : Additional keyword arguments to pass as keywords arguments to the underlying serialization method.
+        path (str) : Location on disk or S3 path to write `data_description.json` and entity data.
+        profile_name (str): The AWS profile specified to write to S3. Default: None
+        kwargs (keywords) : Additional keyword arguments to pass as keywords arguments to the underlying serialization method or to specify AWS profile.
     '''
-    path = os.path.abspath(path)
-    if os.path.exists(path):
-        shutil.rmtree(path)
-    for dirname in [path, os.path.join(path, 'data')]:
-        os.makedirs(dirname)
-    description = entityset_to_description(entityset)
-    for entity in entityset.entities:
-        loading_info = write_entity_data(entity, path, **kwargs)
-        description['entities'][entity.id]['loading_info'].update(loading_info)
-    file = os.path.join(path, 'data_description.json')
-    with open(file, 'w') as file:
-        json.dump(description, file)
+    if _is_s3(path):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            os.makedirs(os.path.join(tmpdir, 'data'))
+            description = entityset_to_description(entityset)
+            for entity in entityset.entities:
+                loading_info = write_entity_data(entity, tmpdir, **kwargs)
+                description['entities'][entity.id]['loading_info'].update(loading_info)
+            file = os.path.join(tmpdir, 'data_description.json')
+            with open(file, 'w') as file:
+                json.dump(description, file)
+
+            file_name = "es-{date:%Y-%m-%d_%H:%M:%S}".format(date=datetime.datetime.now())
+            file_path = os.path.join(tmpdir, file_name)
+            tar = tarfile.open(str(file_path) + ".tar", 'w')
+            tar.add(str(tmpdir) + '/data_description.json', arcname='data_description.json')
+            tar.add(str(tmpdir) + '/data', arcname='/data')
+            tar.close()
+            tar = tarfile.open(str(file_path) + ".tar")
+            if(profile_name is not None):
+                transport_params = {'session': boto3.Session(profile_name=profile_name)}
+                with open(file_path + ".tar", 'rb') as fin:
+                    with open(path, 'wb', transport_params=transport_params) as fout:
+                        for line in fin:
+                            fout.write(line)
+            else:
+                s3 = s3fs.S3FileSystem(anon=True)
+                with open(file_path + ".tar", 'rb') as fin:
+                    with s3.open(path, 'wb') as fout:
+                        for line in fin:
+                            fout.write(line)
+    elif _is_url(path):
+        raise ValueError("Writing to URLs is not supported")
+    else:
+        path = os.path.abspath(path)
+        if os.path.exists(path):
+            shutil.rmtree(path)
+        for dirname in [path, os.path.join(path, 'data')]:
```

Comment:
I think using `os.makedirs` on the joined path will be sufficient

Reply:
Fixed